File: JAVA/academic_people/Student.java
package pessoa_academico;

public class Student extends Person {
    private String courseName;
    private int courseDuration;
    private String registrationId;
    private double grade;
    private boolean dependence;
    
    public Student(){}
    public Student(double grade, String name, int age, String courseName){
        // this.name = name;   //provavelmente a melhor opção é mudar os atributos na classe pai pra protected
        // this.grade = grade; 

        setName(name);  //se os atributos são private o melhor jeito q eu pensei foi usar set pra mudar
        setGrade(grade);
        setCourseName(courseName);
        setAge(age);
    }
    public void setGrade(double grade){
        this.grade = grade;
    }
    public double getGrade(){
        return grade;
    }
    public void setDependence(boolean dependence){
        this.dependence = dependence;
    }
    public boolean getDependence(){
        return dependence;
    }
    public void setCourseName(String courseName) {
        this.courseName = courseName;
    }
    public String getCourseName() {
        return courseName;
    }
    public void setCourseDuration(int courseDuration) {
        this.courseDuration = courseDuration;
    }
    public int getCourseDuration() {
        return courseDuration;
    }
    public void setRegistrationId(String registrationId) {
        this.registrationId = registrationId;
    }
    public String getRegistrationId() {
        return registrationId;
    }
    
    public void statusStudent(Professor professor){
        System.out.println("Name: "+getName());
        System.out.println("Age: "+getAge());
        System.out.println("Course name: "+getCourseName());
        System.out.println("Professor name: "+professor.getName());
        System.out.println("Dependence status: "+ (getDependence()?"Approved":"Repproved"));
        System.out.println("\n\n");
    }

}

File: CPP/pessoa_academico/Student.cpp
#include <iostream>
using namespace std;
#include "header.h"

class Student : public Person{
    private:
        string courseName;
        int courseDuration;
        double grade;
        bool dependence;

    public:
        Student(){}
        Student(double _grade, string name, int age, string _courseName){
            grade = _grade;
            courseName = _courseName;

            setName(name);
            setAge(age);
        }

        void setCourseDuration(int _courseDuration){
            courseDuration = _courseDuration;
        }
        void setDependence(bool _dependence){
            dependence = _dependence;
        }
        void setCourseName(string _courseName){
            courseName = _courseName;
        }
        double getGrade(){
            return grade;
        }
        string getCourseName(){
            return courseName;
        }
        int getCourseDuration(){
            return courseDuration;
        }
        bool getDependece(){
            return dependence;
        }
        
        void statusStudent(Professor professor){
            cout << "Name: " + getName() << endl;
            cout << "Age: " + getAge() << endl;
            cout << "Course name: " + getCourseName() << endl;
            // cout << "Professor name: " + professor.getName() << endl;
            // cout << "Grade: " + getGrade() << endl;
            cout << "Dependence status: ";
                // + (getDependece()?"Approved":"Repproved") << endl;
            if(getDependece())
                cout << "Approved" << endl;
            else
                cout << "Repproved" << endl;

            cout << "------------------" << endl;
    }
};